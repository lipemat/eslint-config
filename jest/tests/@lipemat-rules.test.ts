import jestRunnerEslint from '../helpers/jest-runner-eslint';

describe( '@lipemat rules are enabled', () => {
	test( '@lipemat/no-unsafe-value failing', async () => {
		expect( await jestRunnerEslint( 'failing/@lipemat/security' ) ).toMatchSnapshot();
	} );
} );
