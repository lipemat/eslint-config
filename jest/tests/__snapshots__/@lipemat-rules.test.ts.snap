// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lipemat rules are enabled @lipemat/no-unsafe-value failing 1`] = `
"FAIL jest/fixtures/failing/@lipemat/security/global-sinks.ts

E:\\SVN\\the-boilerplate\\packages\\eslint-config\\jest\\fixtures\\failing\\@lipemat\\security\\global-sinks.ts
  6:1   error  eval argument must be sanitized with sanitize() or DOMPurify.sanitize()                            @lipemat/security/no-unsafe-value
  6:1   error  eval can be harmful                                                                                no-eval
  8:1   error  document.write argument must be sanitized with sanitize() or DOMPurify.sanitize()                  @lipemat/security/no-unsafe-value
  8:10  error  'write' is deprecated. [MDN Reference](https://developer.mozilla.org/docs/Web/API/Document/write)  deprecation/deprecation

✖ 4 problems (4 errors, 0 warnings)

FAIL jest/fixtures/failing/@lipemat/security/not-sanitized.tsx

E:\\SVN\\the-boilerplate\\packages\\eslint-config\\jest\\fixtures\\failing\\@lipemat\\security\\not-sanitized.tsx
   3:5   error  'arbitrary' is never reassigned. Use 'const' instead                                               prefer-const
   5:15  error  's' is defined but never used                                                                      @typescript-eslint/no-unused-vars
   9:10  error  'ReactComponent' is defined but never used                                                         @typescript-eslint/no-unused-vars
  14:9   error  dangerouslySetInnerHTML value must be sanitized with sanitize() or DOMPurify.sanitize()            @lipemat/security/no-unsafe-value
  19:7   error  'Test' is assigned a value but never used                                                          @typescript-eslint/no-unused-vars
  21:2   error  jQuery method after argument must be sanitized with sanitize() or DOMPurify.sanitize()             @lipemat/security/no-unsafe-value
  22:2   error  jQuery method append argument must be sanitized with sanitize() or DOMPurify.sanitize()            @lipemat/security/no-unsafe-value
  23:2   error  jQuery method appendTo argument must be sanitized with sanitize() or DOMPurify.sanitize()          @lipemat/security/no-unsafe-value
  24:2   error  jQuery method before argument must be sanitized with sanitize() or DOMPurify.sanitize()            @lipemat/security/no-unsafe-value
  25:2   error  jQuery method html argument must be sanitized with sanitize() or DOMPurify.sanitize()              @lipemat/security/no-unsafe-value
  26:2   error  jQuery method insertAfter argument must be sanitized with sanitize() or DOMPurify.sanitize()       @lipemat/security/no-unsafe-value
  27:2   error  jQuery method insertBefore argument must be sanitized with sanitize() or DOMPurify.sanitize()      @lipemat/security/no-unsafe-value
  28:2   error  jQuery method prepend argument must be sanitized with sanitize() or DOMPurify.sanitize()           @lipemat/security/no-unsafe-value
  29:2   error  jQuery method prependTo argument must be sanitized with sanitize() or DOMPurify.sanitize()         @lipemat/security/no-unsafe-value
  30:2   error  jQuery method replaceAll argument must be sanitized with sanitize() or DOMPurify.sanitize()        @lipemat/security/no-unsafe-value
  31:2   error  jQuery method replaceWith argument must be sanitized with sanitize() or DOMPurify.sanitize()       @lipemat/security/no-unsafe-value
  36:3   error  Assignment to innerHTML must be sanitized with sanitize() or DOMPurify.sanitize()                  @lipemat/security/no-unsafe-value
  40:8   error  String concatenation with potential HTML detected. Use sanitize() or DOMPurify.sanitize()          @lipemat/security/no-unsafe-value
  40:8   error  'str' is assigned a value but never used                                                           @typescript-eslint/no-unused-vars
  43:2   error  jQuery method html argument must be sanitized with sanitize() or DOMPurify.sanitize()              @lipemat/security/no-unsafe-value
  54:2   error  Assignment to window.name must be sanitized with sanitize() or DOMPurify.sanitize()                @lipemat/security/no-unsafe-value
  55:2   error  Assignment to window.status must be sanitized with sanitize() or DOMPurify.sanitize()              @lipemat/security/no-unsafe-value
  55:9   error  'status' is deprecated. [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/status)  deprecation/deprecation
  68:13  error  'status' is deprecated. [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/status)  deprecation/deprecation

✖ 24 problems (24 errors, 0 warnings)
  1 error and 0 warnings potentially fixable with the \`--fix\` option.

Test Suites: 2 failed, 2 total
Tests:       28 failed, 28 total
Snapshots:   0 total
Time:        
Ran all test suites.
"
`;
